2003-10-08  Jeffrey Stedfast  <fejj@ximian.com>

	* camel-mime-utils.c (header_decode_date): Allow timezone offsets
	to be up to 14 hours ahead of UTC. Fixes bug #49357.

Index: camel-mime-utils.c
===================================================================
RCS file: /cvs/gnome/evolution/camel/camel-mime-utils.c,v
retrieving revision 1.183.4.4
diff -u -r1.183.4.4 camel-mime-utils.c
--- camel/camel-mime-utils.c	26 Sep 2003 16:04:49 -0000	1.183.4.4
+++ camel/camel-mime-utils.c	8 Oct 2003 16:42:55 -0000
@@ -3530,12 +3530,12 @@
 		offset = (*inptr++)=='-'?-1:1;
 		offset = offset * header_decode_int(&inptr);
 		d(printf("abs signed offset = %d\n", offset));
-		if (offset < -1200 || offset > 1200)
+		if (offset < -1200 || offset > 1400)
 			offset = 0;
 	} else if (isdigit(*inptr)) {
 		offset = header_decode_int(&inptr);
 		d(printf("abs offset = %d\n", offset));
-		if (offset < -1200 || offset > 1200)
+		if (offset < -1200 || offset > 1400)
 			offset = 0;
 	} else {
 		char *tz = decode_token(&inptr);
